rnd/autogpt_server/autogpt_server/data/agent_block.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING, Iterator
+
+from autogpt.agents.agent import Agent, AgentSettings
+from autogpt.app.config import ConfigBuilder
+from autogpt_server.data.block import Block, BlockOutput, BlockSchema
+from forge.agent.protocols import (
+    CommandProvider,
+)
+from forge.command import command
+from forge.command.command import Command
+from forge.file_storage import FileStorageBackendName, get_storage
+from forge.file_storage.base import FileStorage
+from forge.llm.providers import (
+    MultiProvider,
+)
+from forge.models.action import ActionErrorResult
+from forge.models.json_schema import JSONSchema
+from pydantic import BaseModel, Field
+
+if TYPE_CHECKING:
+    from autogpt.app.config import AppConfig
+
+logger = logging.getLogger(__name__)
+
+
+class BlockAgentSettings(AgentSettings):
+    disabled_components: list[str] = Field(default_factory=list)
```

Comment:
If we move this to `autogpt` `AgentSettings` then I think we can remove `BlockAgent`